move/move-ibc/sources/dispatch_ibc.move

```diff
@@ -0,0 +1,2444 @@
+            client_state
+        );
+
+        let i = 0;
+        while (i < heights_len) {
+            let height = *vector::borrow(&heights, i);
+
+            table::upsert(
+                &mut store.commitments,
+                commitment::consensus_state_commitment_key(client_id, height),
+                hash::sha2_256(*vector::borrow(&consensus_states, i))
+            );
+
+            event::emit(
+                ClientUpdated {
+                    client_id,
+                    // NOTE: This is currently enforced, if/when we refactor to be more general across clients then this will need to be modified accordingly
+                    client_type: string::utf8(CLIENT_TYPE_COMETBLS),
+                    height
+                }
+            );
+
+            i = i + 1;
+        };
+    }
+
+    public entry fun submit_misbehaviour(
+        client_id: u32, misbehaviour: vector<u8>
+    ) acquires IBCStore {
+        let store = borrow_global_mut<IBCStore>(get_vault_addr());
+
+        assert!(
+            table::contains(
+                &store.commitments,
+                commitment::client_state_commitment_key(client_id)
+            ),
+            E_CLIENT_NOT_FOUND
+        );
+
+        light_client::report_misbehaviour(client_id, misbehaviour);
+
+        event::emit(
+            SubmitMisbehaviour {
+                client_id,
+                client_type: string::utf8(CLIENT_TYPE_COMETBLS)
+            }
+        );
+    }
+
+    public entry fun channel_open_init<T: key + store + drop>(
+        port_id: address,
+        connection_id: u32,
+        ordering: u8,
+        version: vector<u8>
+    ) acquires IBCStore, Port {
+
+        let port = borrow_global<Port<T>>(get_vault_addr());
+        assert!(port.port_id == port_id, E_UNAUTHORIZED);
+
+        let port_id = address_to_string(port_id);
+
+        ensure_connection_state(connection_id);
+
+        let channel_id = generate_channel_identifier();
+
+        let store = borrow_global_mut<IBCStore>(get_vault_addr());
+
+        let channel = channel::default();
+        channel::set_state(&mut channel, CHAN_STATE_INIT);
+        channel::set_ordering(&mut channel, ordering);
+        channel::set_connection_id(&mut channel, connection_id);
+        channel::set_version(&mut channel, version);
+
+        table::upsert(
+            &mut store.commitments,
+            commitment::next_sequence_send_commitment_key(channel_id),
+            bcs::to_bytes(&1)
+        );
+
+        table::upsert(
+            &mut store.commitments,
+            commitment::next_sequence_recv_commitment_key(channel_id),
+            bcs::to_bytes(&1)
+        );
+
+        table::upsert(
+            &mut store.commitments,
+            commitment::next_sequence_ack_commitment_key(channel_id),
+            bcs::to_bytes(&1)
+        );
+
+        commit_channel(channel_id, channel);
+
+        let dynamic_dispatch_param =
```

Comment:
if we are not gonna be able to do this with `Any`, let's have helper functions such as:
```move
new_channel_open_init_params(ordering, connection_id, channel_id, version)
```